### PR TITLE
perf(login)!: Set-up messenger filters outside login flow

### PR DESCRIPTION
**Pull request by ilmotta**
- Root issue https://github.com/status-im/status-mobile/issues/20059
- Related mobile PR https://github.com/status-im/status-mobile/pull/20173

### Summary

This PR fixes the slow login in mobile devices when users have joined large communities, such as the Status one. A user would get stuck for almost 20s in some devices.

We identified that the step to set-up filters in the messenger is potentially expensive and that it is not critical to happen before the `node.login` signal is emitted. The solution presented in this PR is to set-up filters inside `messenger.Start()`, which the client already calls immediately after login.

With this change, users of the mobile app can login pretty fast even when they joined large 
communities. They can immediately interact with other parts of the app even if filter initialization is running in the background, like Wallet, Activity Center, Settings, and Profile.

### But does this PR work?

This PR was in WIP and it was tested sufficiently well that we know the solution solves the problem and doesn't cause regressions. But it will surely go through another round of QA after mobile and status-go PRs are approved.

### Breaking changes

In the mobile code, we had to change where the endpoint `wakuext_startMessenger` was called and the order of a few events to process chats. So essentially ordering, but no data changes. We hope the story for Desktop is similar :)

Status: ready

### CODE REVIEW

File: protocol/messenger_identity.go
Comment (qfrank) [inline protocol/messenger_identity.go:155]: image

**Review by qfrank — APPROVED** [review]
LGTM, well spotted!

File: protocol/messenger_identity.go
Comment (ilmotta) [reply protocol/messenger_identity.go:155]: Thanks @qfrank, that helped me now reach the root commit.

It looks like this tiny problem was introduced by commit https://github.com/status-im/status-go/commit/f87a38074#diff-b0493cab0297f2e6f0866d54b819f48ff079659dd62f68648743b32819ecbbcfR166-R174

@Samyoul, this commit is quite old already, so it's totally expected if you don't remember, but do you know why we are always concatenating `imd.Name` when it's in fact empty? This implementation carried over and it's still present in `develop`. The `Sprintf` line always generates a metadata name with a whitespace at the end, which I caught by chance while unit testing installation initialization.

It's probably harmless, otherwise it would have caused problems.

**Review by osmaczko — APPROVED** [review]
Nice :+1: (thanks for great description). Should this PR be tested on desktop as well before merge?

P.S.
I am wondering whether this commit should be marked as a breaking change. The order of the filters' initialization seems to be an implementation detail, and clients should work without changes. I know it wasn't the case for mobile, but wasn't that more due to incorrect assumptions? (no strong opinion here)

**Review by igor-sirotin — APPROVED** [review]
Looks good, thanks!

**Comment by ilmotta** [comment]
> Nice 👍 (thanks for great description). Should this PR be tested on desktop as well before merge?
> 
> P.S. I am wondering whether this commit should be marked as a breaking change. The order of the filters' initialization seems to be an implementation detail, and clients should work without changes. I know it wasn't the case for mobile, but wasn't that more due to incorrect assumptions? (no strong opinion here)

Thanks for the review @osmaczko 👍🏼 

> I am wondering whether this commit should be marked as a breaking change.

I used the format `perf(login)!:` to mean there's a breaking change. Is there another way to mark the commit as a breaking change?

> The order of the filters' initialization seems to be an implementation detail, and clients should work without changes.

Good question :) You are correct, the changes in this PR shouldn't affect clients ideally. In mobile, we have to be very particular about the order of certain dispatched events after we receive the signal `node.login` for performance reasons. For instance, the endpoint `wakuext_startMessenger` is called immediately after the signal arrives, but because filter initialization now happens when this endpoint is called, we had to delay dispatching some events otherwise some things didn't work as our QAs found out.

> Should this PR be tested on desktop as well before merge?

Yes @osmaczko, I added this note to the status-go and mobile PRs descriptions. We won't merge until you guys give us the green light that no regressions are introduced in desktop.

**Comment by osmaczko** [comment]
> I used the format perf(login)!: to mean there's a breaking change. Is there another way to mark the commit as a breaking change?

I know it was marked as a breaking change; I was only questioning if it should really be marked as such :sweat_smile:

**Review by jrainville — APPROVED** [review]

**Comment by jrainville** [comment]
I created a desktop PR here to have someone test it https://github.com/status-im/status-desktop/pull/15079

cc @anastasiyaig

**Comment by anastasiyaig** [comment]
thanks @jrainville , i am on it

**Comment by anastasiyaig** [comment]
@ilmotta i cant restore account in this PR, it says that account already exists but it is not true . tried both windows and linux
cc @jrainville 


```
DBG 2024-06-07 11:51:54.532+03:00 Account saved succesfully                  topics="accounts-service" tid=4959 file=service.nim:438
ERROR[06-07|11:51:54.538|github.com/status-im/status-go/mobile/status.go:344] failed to restore account                error="account already exists"
```

image


[Screencast from 07.06.2024 11:51:50.webm](https://github.com/status-im/status-go/assets/82375995/d1ca5075-7b9c-4c70-b9f8-fb0205f6fb5b)

**Comment by pavloburykh** [comment]
> @ilmotta i cant restore account in this PR, it says that account already exists but it is not true . tried both windows and linux
> cc @jrainville

Hey @anastasiyaig! Is this only on windows / linux? Cause I was unable to replicate on Mac.

**Comment by anastasiyaig** [comment]
i did not try mac yet , thought it is enough to test win and unix (as linux) to give it a try at least

**Comment by anastasiyaig** [comment]
@pavloburykh i tried on mac , indeed on mac it works. i will keep digging

**Comment by ilmotta** [comment]
Thank you @anastasiyaig. Although it is more probable that the problem you found is related to the code changes in this PR, it could well be that you found a regression that requires changes in status-desktop and not in status-go. We had to do some changes as well the mobile code to adjust, so I wouldn't be surprised if you hit regressions.

I'll stay tuned to your updates.

**Comment by anastasiyaig** [comment]
@ilmotta it is not about your changes, i can reproduce the same problem in release and master builds. I would say it is amazing go 1.21 update :D I logged ticket here https://github.com/status-im/status-desktop/issues/15101 , sorry for false alarm

**Pull request merged**
